server/src/main/java/org/elasticsearch/action/admin/cluster/stats/MappingStats.java

```diff
@@ -86,9 +86,30 @@ public static MappingStats of(Metadata metadata, Runnable ensureNotCancelled) {
                     FieldStats stats;
                     if (type.equals("dense_vector")) {
                         stats = fieldTypes.computeIfAbsent(type, DenseVectorFieldStats::new);
-                        boolean indexed = fieldMapping.containsKey("index") ? (boolean) fieldMapping.get("index") : false;
+                        DenseVectorFieldStats vStats = (DenseVectorFieldStats) stats;
+                        if (fieldMapping.containsKey("similarity")) {
+                            Object similarity = fieldMapping.get("similarity");
+                            vStats.vectorSimilarityTypeCount.compute(similarity.toString(), (t, c) -> c == null ? count : c + count);
+                        }
+                        String elementTypeStr = "float";
+                        if (fieldMapping.containsKey("element_type")) {
+                            Object elementType = fieldMapping.get("element_type");
+                            elementTypeStr = elementType.toString();
+                        }
+                        vStats.vectorElementTypeCount.compute(elementTypeStr, (t, c) -> c == null ? count : c + count);
+                        boolean indexed = fieldMapping.containsKey("index") && (boolean) fieldMapping.get("index");
                         if (indexed) {
-                            DenseVectorFieldStats vStats = (DenseVectorFieldStats) stats;
+                            Object indexOptions = fieldMapping.get("index_options");
+                            // NOTE, while the default for `float` is now `int8_hnsw`, that is actually added to the mapping
+                            // if the value is truly missing & we are indexed, we default to hnsw.
```

Comment:
Under what condition this could happen that the value for `index_options.type` is missing?

Reply:
@mayya-sharipova when we default to `index: true`, but the element type is `byte`. in that scenario we won't provide a value to the mapping and its instead `hnsw`.